probe/proc/cache.go

```diff
@@ -0,0 +1,123 @@
+package proc
+
+import (
+	"bytes"
+	"sync"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+const (
+	filesCacheLen        = 1024
+	filesCacheExpiration = 60 * time.Second
+)
+
+type filesCacheEntry struct {
+	sync.RWMutex
+	file     File
+	contents *bytes.Buffer
+	creation time.Time
+}
+
+func (fce *filesCacheEntry) Pinned() bool { return fce.contents != nil }
```

Comment:
Hmmm, I'm not sure... cache entries are created either with a valid `contents` or with no contents at all, and that does not change during the lifetime of the entry. My only concern is whether this access could be interleaved with the cache entry removal...